content/articles/dnssec-and-secondary-dns.markdown

```diff
@@ -0,0 +1,23 @@
+---
+title: Why DNSSEC and Secondary DNS may not work together
+excerpt: A detailed explanation as to why DNSSEC and Secondary DNS may not be compatible together.
+categories:
+- DNS
+---
+
+### Table of Contents {#toc}
+
+* TOC
+{:toc}
+
+---
+
+## Why won't DNSSEC and Secondary DNS work together?
+
+Here is why DNSSEC and secondary DNS may not work together:
+
+All authoritative name server responses should include the RR set signatures (RRSIG) corresponding to any published DNSKEY that is the Zone Signing Key (ZSK) type. This is necessary because a resolve could get the ZSK from one authoritative and the RRSIG from another authoritative. To include the RRSIG for all ZSK DNSKEYs, the authoritative name servers must have all private key material. We currently do not share private key material, and even if we did, AXFR does not support transferring private key material and thus key rotations would have to be done manually in a coordinate fashion.
```

Comment:
```suggestion
All authoritative name servers MUST sign all record sets with all private keys that are Zone Signing Key (ZSK) type. This is necessary because a resolver could get the ZSK from one authoritative name server and the RRSIG from another authoritative name server. To include the RRSIG for all ZSK DNSKEYs, the authoritative name servers must have all private key material. We currently do not share private key material, and even if we did, AXFR does not support transferring private key material and thus key rotations would have to be done manually in a coordinated fashion.
```